programs/mango-v4/src/instructions/openbook_v2_place_order.rs

```diff
@@ -165,8 +165,11 @@ pub fn openbook_v2_place_order(
         openbook.highest_placed_ask = 0.0;
     }
     // in the normal quote per base units
-    let limit_price =
-        order.max_quote_lots_including_fees as f64 * quote_lot_size as f64 / base_lot_size as f64;
+    let bids = ctx.accounts.bids.load_mut()?;
+    let asks = ctx.accounts.asks.load_mut()?;
+    let order_book = openbook_v2::state::Orderbook { bids, asks };
+    let price_lots = order.price(now_ts, None, &order_book)?.0;
```

Comment:
this depends on the state of the orderbook and we've just modified it by placing an order; I think it'd be safer to figure out price_lots before any orderbook modification happens

also move down the comment "// in the normal quote per base units", it should cover `let limit_price = ...`